pypeit/scripts/coadd_2dspec.py

```diff
@@ -227,11 +242,42 @@ def main(args):
                 sci_dict[coadd.detname]['waveimg'] \
                     = coadd.reduce(pseudo_dict, show=args.show, show_peaks=args.peaks, basename=basename)
 
+            # Tack on detector (similarly to pypeit.extract_one)
+            for sobj in sci_dict[coadd.detname]['specobjs']:
+                sobj.DETECTOR = sci_dict[coadd.detname]['detector']
+                # iwv = np.where(self.caliBrate.wv_calib.spat_ids == sobj.SLITID)[0][0]
+                # sobj.WAVE_RMS =self.caliBrate.wv_calib.wv_fits[iwv].rms
+
+            # fill the specobjs container
+            all_specobjs.add_sobj(sci_dict[coadd.detname]['specobjs'])
+
+            # fill the spec2dobj container but first ...
```

Comment:
Again these lines are confusing. Same comment as in PypeIt. At the very least please put in a comment to indicate what you are doing here and the logic behind it. It is add that you are assigning this thing and then zeroing it out in slits.